### PR TITLE
Week 4 tdd

### PR DESCRIPTION
**Pull request by HeHaowei**

### CODE REVIEW

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,71 @@
...
   }
+
+  private ArrayList<TimeRange> mergeEventTime(ArrayList<TimeRange> event_time) {
+
+      int size = event_time.size();
+      if (size < 2) return event_time;
+      
+      Collections.sort(event_time, TimeRange.ORDER_BY_START);
+      ArrayList<TimeRange> event_time_start_order = new ArrayList<TimeRange>(event_time);
+      Collections.sort(event_time, TimeRange.ORDER_BY_END);
```
Comment (kai-dream) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:73]: is Collections.sort stable?

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,71 @@
...
+      ArrayList<TimeRange> event_time_end_order = new ArrayList<TimeRange>(event_time);
+      ArrayList<TimeRange> result = new ArrayList<TimeRange>();
+
+      for (int i = 0, j = 0; i < size; i++) {
```
Comment (kai-dream) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: It took me a while to understand your algorithm. I suggest you put more comments explaining what you are doing here, and also give more descriptive name to i, and j.

For exampe, j can be called "beginSection", and i can be called "endSection".

**Review by kai-dream — APPROVED** [review]
Generally good. More comments would be better also make easier if someone else needs to maintain your code.

LGTM +1

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (HeHaowei) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Yes sure, thanks for your suggestion and I have updated the comments already. Thanks for checking with that! I will be aware of this in the future.

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (HeHaowei) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:73]: I think it is stable as Collections.sort uses mergeSort and will not change the original relative position of two equal element. But whether or not this sort is stable would not affect the result I think.

Comment (kai-dream) [reply walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:73]: I see. I misunderstood your algorithm. Reading it again, you don't need the 2 sort to be stable, as you are only using one endpoint of the range in each list. 

Looks good me.

**Review by kai-dream — APPROVED** [review]
LGTM +1

**Pull request merged**